python/ray/serve/_private/application_state.py

```diff
@@ -735,10 +735,15 @@ def list_deployment_details(self) -> Dict[str, DeploymentDetails]:
         return {k: v for k, v in details.items() if v is not None}
 
     def _update_status(self, status: ApplicationStatus, status_msg: str = "") -> None:
-        if status_msg and status in [
-            ApplicationStatus.DEPLOY_FAILED,
-            ApplicationStatus.UNHEALTHY,
-        ]:
+        if (
```

Comment:
the application is redeployed with same error, there is no error msg in this case, is this possible scenario?

Reply:
No, the status should aways switch to `DEPLOYING` before it transitions to `DEPLOY_FAILED` with an error message.